### PR TITLE
Handle tables with no ideal state in SegmentStatusChecker.

### PR DESCRIPTION
**Pull request by rven1**

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java [hunk 1]
```diff
@@ -175,6 +175,8 @@ public void runSegmentMetrics() {
       }
       IdealState idealState = helixAdmin.getResourceIdealState(helixClusterName, tableName);
       if (idealState == null) {
```
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: does helix return null or empty set if idealstate is empty ? Null idealstate seems to be error condition.

For empty idealstate, # of up replicas is # of replicas for the table

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/SegmentStatusCheckerTest.java [hunk 2]
```diff
@@ -303,7 +303,9 @@ public void missingIdealTest() throws Exception {
     Assert.assertEquals(controllerMetrics.getValueOfTableGauge(tableName,
         ControllerGauge.SEGMENTS_IN_ERROR_STATE), 0);
     Assert.assertEquals(controllerMetrics.getValueOfTableGauge(tableName,
-        ControllerGauge.NUMBER_OF_REPLICAS), 0);
+        ControllerGauge.NUMBER_OF_REPLICAS), 1);
```
Comment (antumbde) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/SegmentStatusCheckerTest.java]: Need test case for 0 segments in the table. Ideal state exists but number of partitions of the resource are 0.

Comment (rven1) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/SegmentStatusCheckerTest.java]: Added.

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java [hunk 3]
```diff
@@ -175,6 +175,9 @@ public void runSegmentMetrics() {
...
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/SegmentStatusChecker.java]: Change this condition to cover all cases.

idealState == null || idealState.getPartitionSet().isEmpty()

**Pull request merged**